homeassistant/components/melcloud/climate.py

```diff
@@ -0,0 +1,200 @@
+"""Platform for climate integration."""
+from datetime import timedelta
+import logging
+from typing import List, Optional
+
+from pymelcloud import AtaDevice
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    DEFAULT_MAX_TEMP,
+    DEFAULT_MIN_TEMP,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.components.melcloud import MelCloudDevice
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PRECISION_TENTHS, PRECISION_WHOLE, TEMP_CELSIUS
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util.temperature import convert as convert_temperature
+
+from .const import DOMAIN, HVAC_MODE_LOOKUP, HVAC_MODE_REVERSE_LOOKUP, TEMP_UNIT_LOOKUP
+
+ENTITY_ID_FORMAT = f"{DOMAIN}.{{}}"
+SCAN_INTERVAL = timedelta(seconds=60)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+):
+    """Set up MelCloud device climate based on config_entry."""
+    mel_devices = hass.data[DOMAIN].get(entry.entry_id)
+    async_add_entities(
+        [
+            AtaDeviceClimate(mel_device)
+            for mel_device in mel_devices
+            if isinstance(mel_device.device, AtaDevice)
```

Comment:
Using isinstance is generally a bad sign, as it does not allow easy testing (mocking is hard).

Reply:
A quick Google search on `isinstance` seems to agree.

I did a new release of the `pymelcloud` library that switches the return value from `get_devices` to a dict that's laid out like this: `{DeviceType: List[Device]}`. There's no real use case for having the devices in a single list so this seems natural enough. It also skips the filter step completely as an added bonus.